### ISSUE TITLE
Kicked

### ISSUE THREAD

**Issue by ZVH1** [SnowLord7/Gimkit-Hack#4]
Every time I use this a message pops up on the host computer saying ""Usernam" has been removed for cheating", and on my screen it says kicked for hacking. Stop.

**Comment by SnowLord7** [comment]
Make sure you are using the newest version which isn't detected by the
anti-cheat. I am however working on a better solution that permanently
blocks the cheating request from going through.

**Comment by ZVH1** [comment]
Is the newest version the downoald .zip on this website, becuase I downloaded the zip, extracted it, changed the .js to .txt and paste and ran the code in console.

**Comment by FloppyT** [comment]
@SnowLord7 none of your versions fix the anti-cheat. The anti-cheat/cheat prevention is server side. Only valid solution that comes to mind is to slow down your bot

**Comment by SnowLord7** [comment]
Client side does have cheat prevention, decode the websockets and you will
see certain messages sent when cheating. It is called redboat.

**Comment by FloppyT** [comment]
@SnowLord7, #1 I have fully decoded the websockets and #2, it is called blueboat. The client is kicked by the server, and the server sends a non-close message telling you the probably. I haven't checked in a while so it might have changed - but last time I did check - it is sent from the server.

**Comment by SnowLord7** [comment]
The client checks a lot of things, including the DOM and window variables /
functions. For example, if the Exploit function I made was global, it would
be detected and a message would be sent to their server with the data
'redboat'. It alludes to the name of the library they use / made as you
know, blueboat. It seems as if they delay the kicking of the cheater to
prevent them from knowing what triggered the kick. Honestly, I'm not sure
if there is a server side detection as I was able to solve 5 problems per
second without getting kicked, but solving 1 per second with DOM
modifications got me kicked in under three minutes.

**Comment by FloppyT** [comment]
You're right, I'm looking right now. Never noticed redboat - it must be new. It takes approximately 20s for it to check, but I'm pretty sure it is random.

I do know they have server side detection ( or at least they used to have a lot of it ), but I also know that sending questions from 1000 per second to apparently 5 per second do not trigger it. It doesn't like the fast solves - sometimes because it can't handle it and sometimes because it is not ready. Gimkit is getting better at detecting this sort of stuff, so it will only get harder and harder.

**Comment by FloppyT** [comment]
Also it does track the people who hack, getting information like IP, UserAgent, GameMode, Type of hack (`["Exploit", "jQuery", "clickElement"]`), and Name. So If I were you, I would be careful when developing this, and maybe tell the user of the hack to careful too.

**Comment by Cereal-Boy** [comment]
Everyone realizes FloppyT could be a person who works a gimkit, right?

**Comment by SnowLord7** [comment]
He could be, but then he should at least understand how his own system
works, which he does not.

**Comment by FloppyT** [comment]
No, I don't work for Gimkit, nor have I played it in a while. A few months ago I was able to do some crazy stuff, but I haven't explored the new system - which I think includes some sort of authentication, but nothing too difficult.

**Comment by mattGHEW** [comment]
Could you make something that buys upgrades automatically
